packages/jest-fake-timers/src/jestFakeTimers.ts

```diff
@@ -237,6 +237,20 @@ export default class FakeTimers<TimerRef> {
       .forEach(([timerHandle]) => this._runTimerHandle(timerHandle));
   }
 
+  advanceTimersToNextTimer(steps: number = 1): void {
+    const nextExpiry = Array.from(this._timers.values()).reduce(
```

Comment:
Instead of reducing, you can sort all timers by `expiry` and just get the one at the index of `step`

```js
const sortedTimers = Array.from(this._timers.values()).sort((left, right) => left.expiry - right.expiry);

const timerToAdvanceTo = steps > sortedTimers.length ? sortedTimers.pop() : sortedTimers[steps - 1];

if (timerToAdvanceTo) {
  this.advanceTimersByTime(timerToAdvanceTo.expiry - this._now);
}
```

Then you don't need to use recursion for multiple steps

Reply:
we want to use recursion (calling `advanceTimersByTime` multiple times) because in that way we don't miss `setTimeout` inside `setTimeout`
do you want us to replace the recursion anyway?

Reply:
Depends on the behaviour we want (see the comment I added to the test), but I think we want to figure out how far to advance before running anything, then advance that far regardless of added/removed timers as we go.  But maybe not :)

Reply:
But yes, I agree with your conclusion